### ISSUE TITLE
Support multi-platform builds

### ISSUE THREAD

**Issue by edburns** [oracle/weblogic-image-tool#478]
I asked Bing AI about this and it said:

> However, as of my last knowledge update (which was up to 2021), the WebLogic Image Tool itself does not directly support docker buildx build. [The docker buildx command is part of Docker BuildKit, which enables multi-platform builds and improved caching](https://www.docker.com/blog/image-rebase-and-improved-remote-cache-support-in-new-buildkit/)[2](https://www.docker.com/blog/image-rebase-and-improved-remote-cache-support-in-new-buildkit/). It’s a powerful tool for creating multi-architecture images, but it’s not directly integrated with the WebLogic Image Tool.

Can someone please confirm that multi-platform builds are not supported?

Thanks,

Ed

**Comment by ddsharpe** [comment]
This is correct.  the buildx option for Docker builds has not beed added to WebLogic Image Tool.  If its just a matter of using buildx versus the default builder, you can achieve this by setting the command line option `--builder`.  Because the command line options for both docker builders are the same, this should work.  

In order to support multi-architecture builds with a single execution, WebLogic Image Tool would need a significant enhancement in order to provide different JDK and WLS installers to each of the architecture builds.  It is not clear that this is even possible with the Docker framework, but this has not been exhaustively researched as the demand for the feature has not been seen.

**Comment by ddsharpe** [comment]
@edburns Partially implemented in the next release.  The user can select the target build platform, linux/amd64 or linux/arm64 using `--platform`.  While this uses buildx for podman or docker build engines, it does not support building both platforms at the same time.
